main/src/main/java/cgeo/geocaching/activity/AbstractActionBarActivity.java

```diff
@@ -1,20 +1,62 @@
 package cgeo.geocaching.activity;
 
+import cgeo.geocaching.R;
+import cgeo.geocaching.ui.ViewUtils;
+import cgeo.geocaching.utils.Log;
+
 import android.os.Bundle;
+import android.view.View;
+import android.view.ViewGroup;
+import android.view.Window;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.ActionBar;
+import androidx.core.graphics.Insets;
+import androidx.core.view.OnApplyWindowInsetsListener;
+import androidx.core.view.ViewCompat;
+import androidx.core.view.WindowCompat;
+import androidx.core.view.WindowInsetsCompat;
+import androidx.core.view.WindowInsetsControllerCompat;
 
 /**
  * Classes actually having an ActionBar (as opposed to the Dialog activities)
  */
 public class AbstractActionBarActivity extends AbstractActivity {
 
+    protected int calculateInsets = WindowInsetsCompat.Type.statusBars() | WindowInsetsCompat.Type.displayCutout();
+    boolean skipActionBarInsetCalculation = false; // for edge to edge configuration, see onCreate()
+
     @Override
     public void onCreate(final Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         initUpAction();
-    }
 
+        // edge2edge configuration
+        final Window currentWindow = getWindow();
+        final WindowInsetsControllerCompat windowInsetsController = WindowCompat.getInsetsController(currentWindow, currentWindow.getDecorView());
+        windowInsetsController.setSystemBarsBehavior(WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE);
+        ViewCompat.setOnApplyWindowInsetsListener(currentWindow.getDecorView(), new OnApplyWindowInsetsListener() {
+            @Override
+            public @NonNull WindowInsetsCompat onApplyWindowInsets(final @NonNull View v, final @NonNull WindowInsetsCompat insets) {
+                // only apply padding calculations when edge2edge is currently active
+                final Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
+                final boolean isEdgeToEdge = systemBars.top == systemBars.bottom; // seems to be no official API for this
```

Comment:
If I understand this correctly, you want to differentiate between edge-to-edge being active or not. I guess you want to make this differentiation based on the devices SDK version?
--> why not instead enabling edge-to-edge for ALL Android version (down to SDK21). Hence we don't have to support two types of layout in the future

Reply:
My first goal was to mimic current behavior on targetSDK35 without using the opt-out flag, therefore I started that way. I've switched to calling `enableEdgeToEdge()` (needing to update core lib to 1.17 and compileSDK to 36 for this), so it's common ground now for older API levels as well.